### PR TITLE
fix(login): Login page displays "UNKNOWN"

### LINKED ISSUE #439: UI - Login page should display username
**Issue:** When User is logging back into the app, the PIN page display's _WELCOME BACK, UNKNOWN_ instead of displaying the Users name 

**STR**
- Create an account 
- Log out/back in

**Screenshot:**
Screenshot 2023-03-10 at 10 18 52 PM

### PR DESCRIPTION
**Pull request by Flemmli97**
### What this PR does 📖

- Caches the username and during login displays that name instead of "UNKNOWN"
  Username is simply saved in the state ui instance

### Which issue(s) this PR fixes 🔨

- Resolve #439

### CODE REVIEW

File: ui/src/layouts/unlock.rs [hunk 1]
```diff
@@ -22,6 +23,15 @@ use common::{
 
 use crate::AuthPages;
 
+static LAZY_WELCOME_MESSAGE: Lazy<String> = Lazy::new(|| {
+    let state = State::load();
+    let name = match &state.ui.cached_username {
+        Some(name) => name.clone(),
+        None => String::from("UNKNOWN"),
+    };
+    format!("Welcome back, {}", name)
```
Comment (sdwoodbury) [inline ui/src/layouts/unlock.rs]: "Welcome back" should be updated to use `get_local_text`

File: ui/src/layouts/unlock.rs [hunk 2]
```diff
@@ -175,7 +185,7 @@ pub fn UnlockLayout(cx: Scope, page: UseState<AuthPages>, pin: UseRef<String>) -
                             with_validation: Some(pin_validation),
                             with_clear_btn: true,
                             with_label: if STATIC_ARGS.cache_path.exists()
-                                {Some("Welcome back, UNKNOWN")}
+                            {Some(&LAZY_WELCOME_MESSAGE)}
                             else
                                 {Some("Let's choose your password")}, // TODO: Implement this.
```
Comment (sdwoodbury) [inline ui/src/layouts/unlock.rs]: we should also use `get_local_text` for "Let's choose your password"

**Review by sdwoodbury — COMMENTED** [review]
This looks good. But since we're changing the login page, we ought to add translations too.

edit: `Cargo Workflow` failed. Clippy is displeased :grin:

**Comment by phillsatellite** [comment]
This works really well! I noticed now if the username is too long it gets rendered behind the Mascot, I think this could be fixed in a new ticket so we dont need to hold this one up. If anyone thinks different no worries just let me know! 

Screenshot 2023-03-27 at 2 22 11 PM

**Comment by phillsatellite** [comment]
Changes for Stu's comments 👍

**Review by InfamousVague — APPROVED** [review]

**Pull request merged**